test/gtest/layout_transpose.cpp

```diff
@@ -363,6 +331,21 @@ struct LayoutTransposeTest_3D : public ::testing::TestWithParam<std::tuple<uint3
     uint32_t n;
     uint32_t c;
 
+    miopen::Invoker PrepareLayoutTransposeInvoker(const miopen::ExecutionContext& ctx_,
+                                                  miopenDataType_t data_type_,
+                                                  uint32_t n_,
+                                                  uint32_t c_,
+                                                  uint32_t d_,
+                                                  uint32_t h_,
+                                                  uint32_t w_)
+    {
+        TRANSPOSE_SOL transpose_sol(ctx_, data_type_, n_, c_, d_, h_, w_);
+        auto invoker_factory = transpose_sol.MakeBatchedTransposeInvokerFactory();
```

Comment:
convert these lines to a method (e.g., `PrepareInvoker()`) on the class `BatchedTransposeSol`.

Reply:
See discussion on the other one.